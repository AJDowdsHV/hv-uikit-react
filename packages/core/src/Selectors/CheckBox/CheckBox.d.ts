import { StandardProps, CheckboxProps, FormControlProps } from "@material-ui/core";
import { LabelPositions } from "..";

export interface HvCheckBoxProps extends StandardProps<CheckboxProps, HvCheckBoxClassKey> {
  /**
   * The label to be added to the checkbox.
   */
  label?: React.ReactNode;

  /**
   * Whether the selector should use semantic colors
   */
  semantic: boolean;

  /**
   * The position of the checkbox label.
   *  - Accepted values:
   *    --"start",
   *    --"end"
   *  - note: the labelPositions object should be used to set this value.
   */
  labelPlacement?: LabelPositions;
  /**
   * Extra properties passed to the MUI FormControlLabel component.
   */
  formControlLabelProps?: FormControlProps;
}

export type HvCheckBoxClassKey =
  | "container"
  | "disableFocus"
  | "labelTypography"
  | "labelDisabled"
  | "labelEnd"
  | "labelStart"
  | "checkBox"
  | "icon"
  | "truncate";

export default function HvCheckBox(props: HvCheckBoxProps): JSX.Element | null;
