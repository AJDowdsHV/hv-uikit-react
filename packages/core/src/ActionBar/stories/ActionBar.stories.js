import React from "react";
import { makeStyles } from "@material-ui/core";
import { HvActionContainer, HvButton, HvDropDownMenu } from "../..";

export default {
  title: "Patterns/Action Bar/Action Container",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvActionContainer } from '@hv/uikit-react-core/dist'"
  },
  component: HvActionContainer
};

export const Main = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
              Cancel
            </HvButton>
          </div>
        </div>
        <div>
          <HvButton category="ghost" onClick={() => console.log("Help action")}>
            Help
          </HvButton>
        </div>
      </HvActionContainer>
    </div>
  );
};

export const DualAction = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvButton category="ghost" onClick={() => console.log("Cancel action")}>
              Cancel
            </HvButton>
          </div>
        </div>
      </HvActionContainer>
    </div>
  );
};

DualAction.story = {
  parameters: {
    docs: {
      storyDescription: "Showcasing the action bar pattern with only two actions."
    }
  }
};

export const VariedActionBar = () => {
  const useStyles = makeStyles(theme => ({
    root: {
      margin: "auto",
      backgroundColor: theme.hv.palette.atmosphere.atmo1,
      width: "1000px"
    },
    flexBox: {
      display: "flex"
    },
    separator: {
      marginRight: theme.hv.spacing.sm
    }
  }));

  const classes = useStyles();

  return (
    <div className={classes.root}>
      <HvActionContainer id="ActionContainer">
        <div className={classes.flexBox}>
          <div className={classes.separator}>
            <HvButton category="ghost" onClick={() => console.log("Save action")}>
              Save
            </HvButton>
          </div>
          <div>
            <HvDropDownMenu
              onClick={(e, item) => console.log(item.label)}
              dataList={[{ label: "Delete" }, { label: "Update" }]}
            />
          </div>
        </div>
        <div>
          <HvButton category="ghost" onClick={() => console.log("Help action")}>
            Help
          </HvButton>
        </div>
      </HvActionContainer>
    </div>
  );
};

VariedActionBar.story = {
  parameters: {
    docs: {
      storyDescription: "Showcasing the action bar pattern with only two actions."
    }
  }
};
