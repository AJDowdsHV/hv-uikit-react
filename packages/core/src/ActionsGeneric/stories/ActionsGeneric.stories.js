/* eslint-disable react/prop-types */
import React from "react";
import { Add, Delete, Lock, Preview } from "@hv/uikit-react-icons";
import HvActionsGeneric from "..";

export default {
  title: "Components/Actions Generic",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvActionsGeneric } from '@hv/uikit-react-core/dist'",
    dsVersion: "3.2.0",
  },
  component: HvActionsGeneric,
  decorators: [
    (Story) => (
      <div
        style={{
          display: "flex",
          justifyContent: "center",
        }}
      >
        <Story />
      </div>
    ),
  ],
};

export const Main = () => (
  <HvActionsGeneric
    actions={[
      { id: "add", label: "Add", iconCallback: () => <Add /> },
      { id: "delete", label: "Delete", iconCallback: () => <Delete /> },
      { id: "lock", label: "Lock", iconCallback: () => <Lock /> },
      { id: "put", label: "Preview", iconCallback: () => <Preview /> },
    ]}
    maxVisibleActions={2}
  />
);
