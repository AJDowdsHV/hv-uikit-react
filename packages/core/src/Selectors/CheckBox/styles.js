import { outlineStyles } from "../../Focus/styles";

const styles = theme => ({
  container: {
    height: "32px",
    marginRight: 0,
    marginLeft: 0,
    "&:hover": {
      backgroundColor: theme.hv.palette.atmosphere.atmo3
    },
    "&:focus-within": {
      backgroundColor: theme.hv.palette.atmosphere.atmo3,
      "& svg": {
        ...outlineStyles
      }
    }
  },
  disableFocus: {
    outline: "none !important"
  },
  labelTypography: {
    ...theme.hv.typography.normalText,
    "&$labelDisabled": {
      ...theme.hv.typography.placeholderText
    },
    display: "flex"
  },
  labelDisabled: {
    backgroundColor: "transparent",
    cursor: "not-allowed",
    "&:hover": {
      backgroundColor: "transparent",
      cursor: "not-allowed"
    }
  },
  labelEnd: {
    paddingRight: "8px"
  },
  labelStart: {
    paddingLeft: "8px"
  },
  checkBox: {
    padding: 0
  },
  icon: {}
});

export default styles;
