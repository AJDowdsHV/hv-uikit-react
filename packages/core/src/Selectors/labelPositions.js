const labelPositions = Object.freeze({
  start: "start",
  end: "end"
});

export default labelPositions;
