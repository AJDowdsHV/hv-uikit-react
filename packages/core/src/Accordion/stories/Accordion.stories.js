import React, { useState, useMemo } from "react";
import { makeStyles } from "@mui/styles";
import {
  HvAccordion,
  HvListContainer,
  HvListItem,
  HvInput,
  HvButton,
  HvTable,
  HvTableHead,
  HvTableHeader,
  HvTableRow,
  HvTableBody,
  HvTableCell,
  useHvData,
  useHvTableSticky,
} from "../..";

export default {
  title: "Display/Accordion",
  parameters: {
    componentSubtitle: null,
    usage: "import { HvAccordion } from '@hitachivantara/uikit-react-core'",
  },

  component: HvAccordion,
};

export const Main = () => {
  const useStyles = makeStyles(() => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="item1" label="Analytics" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Views</HvListItem>
          <HvListItem>Parameters</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="item2" label="System" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Settings</HvListItem>
          <HvListItem>Network</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="item3" label="Data" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Storage</HvListItem>
          <HvListItem>Memory</HvListItem>
        </HvListContainer>
      </HvAccordion>
    </div>
  );
};

export const Disabled = () => {
  const useStyles = makeStyles(() => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="disabled-item1" label="Analytics" headingLevel={3} disabled>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Views</HvListItem>
          <HvListItem>Parameters</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="disabled-item2" label="System" headingLevel={3}>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Settings</HvListItem>
          <HvListItem>Network</HvListItem>
        </HvListContainer>
      </HvAccordion>
      <HvAccordion id="disabled-item3" label="Data" headingLevel={3} disabled>
        <HvListContainer className={classes.accordionSpacing} interactive condensed>
          <HvListItem>Storage</HvListItem>
          <HvListItem>Memory</HvListItem>
        </HvListContainer>
      </HvAccordion>
    </div>
  );
};

Disabled.parameters = {
  docs: {
    description: { story: "A common accordion with disable items. " },
  },
};

export const Controlled = () => {
  const useStyles = makeStyles((theme) => ({
    accordionSpacing: {
      "& > li": {
        paddingLeft: 32,
      },
    },
    accordionContainer: {
      maxWidth: 300,
    },
    buttonContainer: {
      "& button": {
        marginRight: theme.hvSpacing("xs"),
        marginBottom: theme.hvSpacing("md"),
      },
    },
  }));
  const classes = useStyles();
  const [expandedState, setExpandedState] = useState({ analytics: true, system: true, data: true });
  const handleToggle = (key) => {
    const newValue = { ...expandedState };
    newValue[key] = !newValue[key];
    setExpandedState(newValue);
  };
  const handleAll = (option) => {
    setExpandedState({ analytics: option, system: option, data: option });
  };
  return (
    <>
      <div className={classes.buttonContainer}>
        <HvButton category="secondary" onClick={() => handleToggle("analytics")}>
          Toggle Analytics
        </HvButton>
        <HvButton category="secondary" onClick={() => handleToggle("system")}>
          Toggle System
        </HvButton>
        <HvButton category="secondary" onClick={() => handleToggle("data")}>
          Toggle Data
        </HvButton>
        <HvButton category="secondary" onClick={() => handleAll(false)}>
          Close all
        </HvButton>
        <HvButton category="secondary" onClick={() => handleAll(true)}>
          Expand all
        </HvButton>
      </div>
      <div className={classes.accordionContainer}>
        <HvAccordion
          id="controlled-item1"
          label="Analytics"
          onChange={() => handleToggle("analytics")}
          expanded={expandedState.analytics}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Views</HvListItem>
            <HvListItem>Parameters</HvListItem>
          </HvListContainer>
        </HvAccordion>
        <HvAccordion
          id="controlled-item2"
          label="System"
          onChange={() => handleToggle("system")}
          expanded={expandedState.system}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Settings</HvListItem>
            <HvListItem>Network</HvListItem>
          </HvListContainer>
        </HvAccordion>
        <HvAccordion
          id="controlled-item3"
          label="Data"
          onChange={() => handleToggle("data")}
          expanded={expandedState.data}
        >
          <HvListContainer className={classes.accordionSpacing} interactive condensed>
            <HvListItem>Storage</HvListItem>
            <HvListItem>Memory</HvListItem>
          </HvListContainer>
        </HvAccordion>
      </div>
    </>
  );
};

Controlled.parameters = {
  docs: {
    description: { story: "An accordion controlled externally forcing expansion." },
  },
};

export const Form = () => {
  const useStyles = makeStyles(() => ({
    accordionContainer: {
      maxWidth: 300,
    },
    formContainer: {
      padding: "0 32px",
      "& > div": {
        paddingTop: 17,
      },
    },
  }));
  const classes = useStyles();
  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="personal" label="Personal Information" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-name" label="Name" placeholder="Insert first name" required />
          <HvInput id="input-email" label="Email" placeholder="Insert your email" required />
          <HvInput id="input-phone" label="Phone" placeholder="Insert your phone number" />
          <HvInput id="input-extension" label="Extension" placeholder="Insert phone extension" />
          <HvInput id="input-country" label="Country" placeholder="Insert country name" />
          <HvInput id="input-province" label="City/Province" placeholder="Insert province name" />
        </div>
      </HvAccordion>
      <HvAccordion id="billing" label="Billing Address" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-address" label="Address 1" placeholder="Insert first name" />
          <HvInput id="input-address2" label="Address 2" placeholder="Insert address" />
          <HvInput id="input-city" label="City" placeholder="Insert city name" />
          <HvInput id="input-state" label="State" placeholder="Insert state" />
          <HvInput id="input-code" label="Zip Code" placeholder="Insert code" />
        </div>
      </HvAccordion>
      <HvAccordion id="shipping" label="Shipping Address" headingLevel={3}>
        <div className={classes.formContainer}>
          <HvInput id="input-address-bill" label="Address 1" placeholder="Insert first name" />
          <HvInput id="input-address2-bill" label="Address 2" placeholder="Insert address" />
          <HvInput id="input-city-bill" label="City" placeholder="Insert city name" />
          <HvInput id="input-state-bill" label="State" placeholder="Insert state" />
          <HvInput id="input-code-bill" label="Zip Code" placeholder="Insert code" />
        </div>
      </HvAccordion>
    </div>
  );
};

Form.parameters = {
  docs: {
    description: { story: "An accordion composition with forms." },
  },
};

export const Typography = () => {
  const useStyles = makeStyles(() => ({
    accordionContainer: {
      width: "100%",
    },
  }));
  const classes = useStyles();

  const getData = () => [
    {
      id: 1,
      title: "A new hope",
      director: "George Lucas",
      producer: "Gary Kurzt",
      release: "1977-05-25",
    },
    {
      id: 2,
      title: "Empires strike back",
      director: "Irvin Kershner",
      producer: "Gary Kurzt",
      release: "1980-05-17",
    },
    {
      id: 3,
      title: "Return of the jedi",
      director: "Richard Marquand",
      producer: "Howard Kazanjian",
      release: "1983-05-25",
    },
  ];

  const getColumns = () => [
    {
      Header: "Title",
      accessor: "title",
      sticky: "left",
      width: 150,
      style: { textTransform: "capitalize" },
    },
    {
      Header: "Director",
      accessor: "director",
    },
    {
      Header: "Producer",
      accessor: "producer",
    },
    {
      Header: "Release Date",
      accessor: "release",
    },
  ];

  const data = useMemo(() => getData(), []);
  const columns = useMemo(() => getColumns(), []);

  const { getTableProps, getTableHeadProps, getTableBodyProps, prepareRow, headerGroups, rows } =
    useHvData(
      {
        columns,
        data,
        stickyHeader: true,
      },
      useHvTableSticky
    );

  return (
    <div className={classes.accordionContainer}>
      <HvAccordion id="film" label="Films" headingLevel={2}>
        <HvTable id="accordion-table" {...getTableProps()}>
          <HvTableHead {...getTableHeadProps()}>
            {headerGroups.map((headerGroup) => (
              <HvTableRow {...headerGroup.getHeaderGroupProps()}>
                {headerGroup.headers.map((col) => (
                  <HvTableHeader {...col.getHeaderProps()}>{col.render("Header")}</HvTableHeader>
                ))}
              </HvTableRow>
            ))}
          </HvTableHead>
          <HvTableBody {...getTableBodyProps()}>
            {rows.map((row) => {
              prepareRow(row);

              return (
                <HvTableRow {...row.getRowProps()}>
                  {row.cells.map((cell) => (
                    <HvTableCell {...cell.getCellProps()}>{cell.render("Cell")}</HvTableCell>
                  ))}
                </HvTableRow>
              );
            })}
          </HvTableBody>
        </HvTable>
      </HvAccordion>
    </div>
  );
};

Typography.parameters = {
  docs: {
    description: { story: "An accordion with a different button typography." },
  },
};
