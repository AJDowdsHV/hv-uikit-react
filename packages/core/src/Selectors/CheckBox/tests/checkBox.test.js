/* eslint-env jest */

// import { mount } from "enzyme";
import React from "react";
import { mount } from "enzyme";

import { Checkbox } from "@hv/uikit-react-icons";

import { HvCheckBox, HvProvider } from "../../..";
import labelPositions from "../../labelPositions";

import { Main, Disabled } from "../stories/CheckBox.stories";

describe("[v3] CheckBox", () => {
  let wrapper;

  const labelStartClassName = "labelStart";
  const labelEndClassName = "labelEnd";

  beforeEach(async () => {
    wrapper = mount(
      <HvProvider>
        <Main />
      </HvProvider>
    );
  });

  it("should be defined", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render correctly", () => {
    expect(wrapper.find(HvCheckBox)).toMatchSnapshot();
  });

  it("should render the Checkbox component", () => {
    const buttonComponent = wrapper.find(HvCheckBox);
    expect(buttonComponent.length).toBe(1);
  });

  it("should have an icon", () => {
    const component = wrapper.find(HvCheckBox);

    expect(component.find(Checkbox).length).toBe(1);
  });

  it("should have an icon when disabled", () => {
    wrapper = mount(
      <HvProvider>
        <Disabled />
      </HvProvider>
    );

    const component = wrapper.find(HvCheckBox);

    expect(component.find(Checkbox).length).toBe(1);
  });

  it("should apply the correct class name when there is a label at the start", () => {
    const mountWrapper = mount(
      <HvProvider>
        <HvCheckBox label="test" labelPlacement={labelPositions.start} />
      </HvProvider>
    ).find(HvCheckBox);
    expect(mountWrapper.html().includes(labelStartClassName)).toBe(true);
    expect(mountWrapper.html().includes(labelEndClassName)).toBe(false);
  });

  it("should apply the correct class name when there is a label at the end", () => {
    const mountWrapper = mount(
      <HvProvider>
        <HvCheckBox label="test" labelPlacement={labelPositions.end} />
      </HvProvider>
    ).find(HvCheckBox);
    expect(mountWrapper.html().includes(labelStartClassName)).toBe(false);
    expect(mountWrapper.html().includes(labelEndClassName)).toBe(true);
  });

  it("should not apply any class name when there is no label specified", () => {
    const mountWrapper = mount(
      <HvProvider>
        <HvCheckBox labelPlacement={labelPositions.start} />
      </HvProvider>
    ).find(HvCheckBox);
    expect(mountWrapper.html().includes(labelStartClassName)).toBe(false);
    expect(mountWrapper.html().includes(labelEndClassName)).toBe(false);
  });
});
