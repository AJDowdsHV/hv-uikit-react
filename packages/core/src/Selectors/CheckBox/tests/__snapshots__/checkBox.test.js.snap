// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[v3] CheckBox should render correctly 1`] = `
<WithStyles(HvCheckbox)
  id="test"
  inputProps={
    Object {
      "aria-label": "Checkbox example",
    }
  }
>
  <HvCheckbox
    classes={
      Object {
        "checkBox": "HvCheckBox-checkBox",
        "container": "HvCheckBox-container",
        "disableFocus": "HvCheckBox-disableFocus",
        "disabledBorder": "HvCheckBox-disabledBorder",
        "icon": "HvCheckBox-icon",
        "labelDisabled": "HvCheckBox-labelDisabled",
        "labelEnd": "HvCheckBox-labelEnd",
        "labelStart": "HvCheckBox-labelStart",
        "labelTypography": "HvCheckBox-labelTypography",
      }
    }
    id="test"
    inputProps={
      Object {
        "aria-label": "Checkbox example",
      }
    }
  >
    <WithStyles(ForwardRef(FormControlLabel))
      className="HvCheckBox-container"
      classes={
        Object {
          "disabled": "HvCheckBox-labelDisabled",
          "label": "HvCheckBox-labelTypography",
        }
      }
      control={
        <ForwardRef(WithStyles)
          checkedIcon={
            <CheckboxCheck
              className="HvCheckBox-icon"
            />
          }
          className="HvCheckBox-checkBox"
          color="default"
          disableRipple={true}
          disabled={false}
          icon={
            <Checkbox
              className="HvCheckBox-icon"
            />
          }
          id="test-input"
          indeterminateIcon={
            <CheckboxPartial
              className="HvCheckBox-icon"
            />
          }
          inputProps={
            Object {
              "aria-label": "Checkbox example",
            }
          }
          onBlur={[Function]}
          onChange={[Function]}
          value=""
        />
      }
      disabled={false}
      id="test"
      labelPlacement="end"
    >
      <ForwardRef(FormControlLabel)
        className="HvCheckBox-container"
        classes={
          Object {
            "disabled": "Mui-disabled HvCheckBox-labelDisabled",
            "label": "MuiFormControlLabel-label HvCheckBox-labelTypography",
            "labelPlacementBottom": "MuiFormControlLabel-labelPlacementBottom",
            "labelPlacementStart": "MuiFormControlLabel-labelPlacementStart",
            "labelPlacementTop": "MuiFormControlLabel-labelPlacementTop",
            "root": "MuiFormControlLabel-root",
          }
        }
        control={
          <ForwardRef(WithStyles)
            checkedIcon={
              <CheckboxCheck
                className="HvCheckBox-icon"
              />
            }
            className="HvCheckBox-checkBox"
            color="default"
            disableRipple={true}
            disabled={false}
            icon={
              <Checkbox
                className="HvCheckBox-icon"
              />
            }
            id="test-input"
            indeterminateIcon={
              <CheckboxPartial
                className="HvCheckBox-icon"
              />
            }
            inputProps={
              Object {
                "aria-label": "Checkbox example",
              }
            }
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          />
        }
        disabled={false}
        id="test"
        labelPlacement="end"
      >
        <label
          className="MuiFormControlLabel-root HvCheckBox-container"
          id="test"
        >
          <WithStyles(ForwardRef(Checkbox))
            checkedIcon={
              <CheckboxCheck
                className="HvCheckBox-icon"
              />
            }
            className="HvCheckBox-checkBox"
            color="default"
            disableRipple={true}
            disabled={false}
            icon={
              <Checkbox
                className="HvCheckBox-icon"
              />
            }
            id="test-input"
            indeterminateIcon={
              <CheckboxPartial
                className="HvCheckBox-icon"
              />
            }
            inputProps={
              Object {
                "aria-label": "Checkbox example",
              }
            }
            onBlur={[Function]}
            onChange={[Function]}
            value=""
          >
            <ForwardRef(Checkbox)
              checkedIcon={
                <CheckboxCheck
                  className="HvCheckBox-icon"
                />
              }
              className="HvCheckBox-checkBox"
              classes={
                Object {
                  "checked": "Mui-checked",
                  "colorPrimary": "MuiCheckbox-colorPrimary",
                  "colorSecondary": "MuiCheckbox-colorSecondary",
                  "disabled": "Mui-disabled",
                  "indeterminate": "MuiCheckbox-indeterminate",
                  "root": "MuiCheckbox-root",
                }
              }
              color="default"
              disableRipple={true}
              disabled={false}
              icon={
                <Checkbox
                  className="HvCheckBox-icon"
                />
              }
              id="test-input"
              indeterminateIcon={
                <CheckboxPartial
                  className="HvCheckBox-icon"
                />
              }
              inputProps={
                Object {
                  "aria-label": "Checkbox example",
                }
              }
              onBlur={[Function]}
              onChange={[Function]}
              value=""
            >
              <WithStyles(ForwardRef(SwitchBase))
                checkedIcon={
                  <CheckboxCheck
                    className="HvCheckBox-icon"
                    fontSize="default"
                  />
                }
                className="HvCheckBox-checkBox"
                classes={
                  Object {
                    "checked": "Mui-checked",
                    "disabled": "Mui-disabled",
                    "root": "MuiCheckbox-root",
                  }
                }
                color="default"
                disableRipple={true}
                disabled={false}
                icon={
                  <Checkbox
                    className="HvCheckBox-icon"
                    fontSize="default"
                  />
                }
                id="test-input"
                inputProps={
                  Object {
                    "aria-label": "Checkbox example",
                    "data-indeterminate": false,
                  }
                }
                onBlur={[Function]}
                onChange={[Function]}
                type="checkbox"
                value=""
              >
                <ForwardRef(SwitchBase)
                  checkedIcon={
                    <CheckboxCheck
                      className="HvCheckBox-icon"
                      fontSize="default"
                    />
                  }
                  className="HvCheckBox-checkBox"
                  classes={
                    Object {
                      "checked": "PrivateSwitchBase-checked-6 Mui-checked",
                      "disabled": "PrivateSwitchBase-disabled-7 Mui-disabled",
                      "input": "PrivateSwitchBase-input-8",
                      "root": "PrivateSwitchBase-root-5 MuiCheckbox-root",
                    }
                  }
                  color="default"
                  disableRipple={true}
                  disabled={false}
                  icon={
                    <Checkbox
                      className="HvCheckBox-icon"
                      fontSize="default"
                    />
                  }
                  id="test-input"
                  inputProps={
                    Object {
                      "aria-label": "Checkbox example",
                      "data-indeterminate": false,
                    }
                  }
                  onBlur={[Function]}
                  onChange={[Function]}
                  type="checkbox"
                  value=""
                >
                  <WithStyles(ForwardRef(IconButton))
                    className="PrivateSwitchBase-root-5 MuiCheckbox-root HvCheckBox-checkBox"
                    color="default"
                    component="span"
                    disableRipple={true}
                    disabled={false}
                    onBlur={[Function]}
                    onFocus={[Function]}
                    tabIndex={null}
                  >
                    <ForwardRef(IconButton)
                      className="PrivateSwitchBase-root-5 MuiCheckbox-root HvCheckBox-checkBox"
                      classes={
                        Object {
                          "colorInherit": "MuiIconButton-colorInherit",
                          "colorPrimary": "MuiIconButton-colorPrimary",
                          "colorSecondary": "MuiIconButton-colorSecondary",
                          "disabled": "Mui-disabled",
                          "edgeEnd": "MuiIconButton-edgeEnd",
                          "edgeStart": "MuiIconButton-edgeStart",
                          "label": "MuiIconButton-label",
                          "root": "MuiIconButton-root",
                          "sizeSmall": "MuiIconButton-sizeSmall",
                        }
                      }
                      color="default"
                      component="span"
                      disableRipple={true}
                      disabled={false}
                      onBlur={[Function]}
                      onFocus={[Function]}
                      tabIndex={null}
                    >
                      <WithStyles(ForwardRef(ButtonBase))
                        centerRipple={true}
                        className="MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root HvCheckBox-checkBox"
                        component="span"
                        disableRipple={true}
                        disabled={false}
                        focusRipple={true}
                        onBlur={[Function]}
                        onFocus={[Function]}
                        tabIndex={null}
                      >
                        <ForwardRef(ButtonBase)
                          centerRipple={true}
                          className="MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root HvCheckBox-checkBox"
                          classes={
                            Object {
                              "disabled": "Mui-disabled",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiButtonBase-root",
                            }
                          }
                          component="span"
                          disableRipple={true}
                          disabled={false}
                          focusRipple={true}
                          onBlur={[Function]}
                          onFocus={[Function]}
                          tabIndex={null}
                        >
                          <span
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-5 MuiCheckbox-root HvCheckBox-checkBox"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            tabIndex={null}
                          >
                            <span
                              className="MuiIconButton-label"
                            >
                              <input
                                aria-label="Checkbox example"
                                className="PrivateSwitchBase-input-8"
                                data-indeterminate={false}
                                disabled={false}
                                id="test-input"
                                onChange={[Function]}
                                type="checkbox"
                                value=""
                              />
                              <Checkbox
                                className="HvCheckBox-icon"
                                fontSize="default"
                              >
                                <WithStyles(HvIconBase)
                                  className="HvCheckBox-icon"
                                  fontSize="default"
                                  iconSize="S"
                                  name="Checkbox"
                                >
                                  <HvIconBase
                                    className="HvCheckBox-icon"
                                    classes={
                                      Object {
                                        "l": "HvIconBase-l",
                                        "m": "HvIconBase-m",
                                        "root": "HvIconBase-root",
                                        "s": "HvIconBase-s",
                                        "xs": "HvIconBase-xs",
                                      }
                                    }
                                    fontSize="default"
                                    iconSize="S"
                                    name="Checkbox"
                                  >
                                    <div
                                      className="HvCheckBox-icon HvIconBase-root HvIconBase-s"
                                      fontSize="default"
                                      name="Checkbox"
                                    >
                                      <svg
                                        focusable={false}
                                        height={16}
                                        viewBox="0 0 16 16"
                                        width={16}
                                        xmlns="http://www.w3.org/2000/svg"
                                      >
                                        <path
                                          className="color0"
                                          d="M0 0h16v16H0z"
                                          fill="#FBFCFC"
                                        />
                                        <path
                                          className="color1"
                                          d="M16 16H0V0h16zM1 15h14V1H1z"
                                          fill="#414141"
                                        />
                                      </svg>
                                    </div>
                                  </HvIconBase>
                                </WithStyles(HvIconBase)>
                              </Checkbox>
                            </span>
                          </span>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(IconButton)>
                  </WithStyles(ForwardRef(IconButton))>
                </ForwardRef(SwitchBase)>
              </WithStyles(ForwardRef(SwitchBase))>
            </ForwardRef(Checkbox)>
          </WithStyles(ForwardRef(Checkbox))>
          <WithStyles(ForwardRef(Typography))
            className="MuiFormControlLabel-label HvCheckBox-labelTypography"
            component="span"
          >
            <ForwardRef(Typography)
              className="MuiFormControlLabel-label HvCheckBox-labelTypography"
              classes={
                Object {
                  "alignCenter": "MuiTypography-alignCenter",
                  "alignJustify": "MuiTypography-alignJustify",
                  "alignLeft": "MuiTypography-alignLeft",
                  "alignRight": "MuiTypography-alignRight",
                  "body1": "MuiTypography-body1",
                  "body2": "MuiTypography-body2",
                  "button": "MuiTypography-button",
                  "caption": "MuiTypography-caption",
                  "colorError": "MuiTypography-colorError",
                  "colorInherit": "MuiTypography-colorInherit",
                  "colorPrimary": "MuiTypography-colorPrimary",
                  "colorSecondary": "MuiTypography-colorSecondary",
                  "colorTextPrimary": "MuiTypography-colorTextPrimary",
                  "colorTextSecondary": "MuiTypography-colorTextSecondary",
                  "displayBlock": "MuiTypography-displayBlock",
                  "displayInline": "MuiTypography-displayInline",
                  "gutterBottom": "MuiTypography-gutterBottom",
                  "h1": "MuiTypography-h1",
                  "h2": "MuiTypography-h2",
                  "h3": "MuiTypography-h3",
                  "h4": "MuiTypography-h4",
                  "h5": "MuiTypography-h5",
                  "h6": "MuiTypography-h6",
                  "noWrap": "MuiTypography-noWrap",
                  "overline": "MuiTypography-overline",
                  "paragraph": "MuiTypography-paragraph",
                  "root": "MuiTypography-root",
                  "srOnly": "MuiTypography-srOnly",
                  "subtitle1": "MuiTypography-subtitle1",
                  "subtitle2": "MuiTypography-subtitle2",
                }
              }
              component="span"
            >
              <span
                className="MuiTypography-root MuiFormControlLabel-label HvCheckBox-labelTypography MuiTypography-body1"
              />
            </ForwardRef(Typography)>
          </WithStyles(ForwardRef(Typography))>
        </label>
      </ForwardRef(FormControlLabel)>
    </WithStyles(ForwardRef(FormControlLabel))>
  </HvCheckbox>
</WithStyles(HvCheckbox)>
`;
